VariantEncoding.md

```diff
@@ -39,13 +39,41 @@ Another motivation for the representation is that (aside from metadata) each nes
 For example, in a Variant containing an Array of Variant values, the representation of an inner Variant value, when paired with the metadata of the full variant, is itself a valid Variant.
 
 This document describes the Variant Binary Encoding scheme.
-[VariantShredding.md](VariantShredding.md) describes the details of the Variant shredding scheme.
+The [Variant Shredding specification](VariantShredding.md) describes the details of shredding Variant values as typed Parquet columns.
+
+## Variant in Parquet
 
-# Variant in Parquet
 A Variant value in Parquet is represented by a group with 2 fields, named `value` and `metadata`.
-Both fields `value` and `metadata` are of type `binary`, and cannot be `null`.
 
-# Metadata encoding
+* The Variant group must be annotated with the `VARIANT` logical type.
```

Comment:
Size limit of what?

Reply:
When Spark implements the Variant, internally it limits value and metadata to 16MB so it will have the limit on how much data can be stored. Different engines may have different limits internally and some Variant values may not be loaded across the engines. I'm wondering if we should have such limit in the spec explicitly. But of course it's really hard to really define such kind of limit since the input data of same size may be encoded differently.

Reply:
I think engines are always going to have different limits on field sizes, I think providing a recommendation on size makes sense but I don't think it should be a rquirement.

Reply:
I agree with Micah and Gene. Let's not put a limit in the spec. That should be determined in other ways.